app/helpers/view_components/view_component.rb

```diff
@@ -1,22 +1,23 @@
 module ViewComponents
   class ViewComponent
-    include ActionView::Helpers::TagHelper
-    include ActionView::Helpers::UrlHelper
-    include IconsHelper
-
-    # TODO: This can probably be removed
-    include ActionView::Helpers::AssetTagHelper
-
-    include ActionView::Context
-
     extend Mandate::Memoize
     extend Mandate::InitializerInjector
 
-    def react_component(id, data)
-      tag.div("", {
-                "data-react-#{id}": true,
-                "data-react-data": data.to_json
-              })
+    delegate :current_user,
+      :safe_join,
+      :tag, :link_to, :button_to,
+      :graphical_icon, :icon, :track_icon, :exercise_icon,
+      to: :view_context
+
+    # This is called when you called `render SomeComponent.new(...)`
+    # It sets the view context, which can then be used for things
+    # like getting the current user, or authentication tokens
+    def render_in(context, *_args)
+      @view_context = context
+      to_s
```

Comment:
This whole PR is basically to add these lines.

Reply:
Didn't know this. Nice!

Reply:
Me neither. Reading the Rails source code++